pkg/securityscan/scanMetricsHandler.go

```diff
@@ -50,11 +43,33 @@ func (c *Controller) handleClusterScanMetrics(ctx context.Context) error {
 		c.numTestsNA.WithLabelValues(scanName, scanProfileName).Set(numTestsNA)
 
 		logrus.Debugf("Done updating metrics for scan %v", obj.Name)
-		v1.ClusterScanConditionAlerted.True(obj)
-		_, err := scans.UpdateStatus(obj)
-		if err != nil {
-			return obj, fmt.Errorf("Retrying, got error %v in updating condition of scan object: %v ", err, obj.Name)
+
+		if obj.Spec.ScheduledScanConfig != nil {
+			updateErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				var err error
```

Comment:
I think RetryOnConflict needs to refetch the object each time and then update it https://godoc.org/k8s.io/client-go/util/retry#RetryOnConflict

Reply:
Good catch, updated!